src/rust-crypto/DehydratedDeviceManager.ts

```diff
@@ -166,19 +166,26 @@ export class DehydratedDeviceManager {
      *
      * @returns the key, if available, or `null` if no key is available
      */
-    private async getKey(create: boolean): Promise<Uint8Array | null> {
-        if (this.key === undefined) {
+    private async getKey(create: boolean): Promise<RustSdkCryptoJs.DehydratedDeviceKey | null> {
+        const cachedKey = await this.getCachedKey();
+        if (!cachedKey) {
```

Comment:
suggest an early return here, to avoid two calls to `getCachedKey` and dodgy `!` operators

```suggestion
        const cachedKey = await this.getCachedKey();
        if (cachedKey) return cachedKey;
```

(can we then inline `getCachedKey`?)